ui/src/app/core/interceptors/httpError.interceptor.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  HttpEvent,
+  HttpInterceptor,
+  HttpHandler,
+  HttpRequest,
+  HttpErrorResponse
+ } from '@angular/common/http';
+ import { Observable, throwError } from 'rxjs';
+ import { retry, catchError } from 'rxjs/operators';
+import { Router } from '@angular/router';
+import { Injectable } from '@angular/core';
+import { ExceptionService } from '../services/exception.service';
+import { capitalize } from '../utils';
+import { ResponseStatusService } from '../services/response-status.service';
+
+@Injectable()
+export class HttpErrorInterceptor implements HttpInterceptor {
+
+  constructor(
+    private exceptionService: ExceptionService,
+    private router: Router,
+    private responseStatusService: ResponseStatusService
+  ) {}
+
+  intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    return next.handle(request).pipe(
+        retry(1),
+        catchError((error: HttpErrorResponse) => {
+          if (error.status !== 200) {
```

Comment:
This approach using interceptor break some existing logic. In some cases the services check the http status and return null if a 404 status is returned. Try search: e.status in the code. In the interface you can go to circulation checkinout and enter a non existing barcode.

Reply:
Intercept http error equal or greater than 400 except 404